ASTProcessor/src/test/java/edu/usb/argos/astprocessor/analyzer/codeSmells/AntlrMethodNormalizerTest.java

```diff
@@ -0,0 +1,104 @@
+    }
+
+    private List<JavaParser.MethodDeclarationContext> parseMethods(String code) {
+        try {
+            CharStream input = CharStreams.fromString(code);
+            JavaLexer lexer = new JavaLexer(input);
+            CommonTokenStream tokens = new CommonTokenStream(lexer);
+            JavaParser parser = new JavaParser(tokens);
+            JavaParser.CompilationUnitContext tree = parser.compilationUnit();
+
+            List<JavaParser.MethodDeclarationContext> methodDeclarations = new ArrayList<>();
+
+            for (JavaParser.TypeDeclarationContext typeDecl : tree.typeDeclaration()) {
+                if (typeDecl.classDeclaration() != null) {
+                    JavaParser.ClassBodyContext classBody = typeDecl.classDeclaration().classBody();
+                    for (JavaParser.ClassBodyDeclarationContext bodyDecl : classBody.classBodyDeclaration()) {
+                        JavaParser.MemberDeclarationContext memberDecl = bodyDecl.memberDeclaration();
+                        if (memberDecl != null) {
+                            if (memberDecl.methodDeclaration() != null) {
+                                methodDeclarations.add(memberDecl.methodDeclaration());
+                            } else if (memberDecl.genericMethodDeclaration() != null) {
+                                methodDeclarations.add(memberDecl.genericMethodDeclaration().methodDeclaration());
+                            }
+                        }
+                    }
+                }
+            }
+
+            return methodDeclarations;
+        } catch (Exception e) {
+            throw new RuntimeException("Error parsing Java code: " + e.getMessage(), e);
+        }
+    }
+
+    @Test
+    public void testNormalizeMethodAsTokens() {
+        String code = """
+                public class MyClass {
+                    public void method5() {
+                        for (int i = 0; i < 20; i++) { // 15
+                            if (i > 10) {
+                                System.out.println(i);
+                                i += 20;
+                                break;
+                            }
+                
+                            method6();
+                            System.out.println("Loop iteration: " + i);
+                            System.out.println(method3("some name"));
+                        }
+                    }
+                }
+                """;
+
+        List<JavaParser.MethodDeclarationContext> methods = parseMethods(code);
+        assertNotNull(methods, "Method context should not be null");
+
+        Optional<JavaParser.MethodDeclarationContext> method = Optional.ofNullable(methods.get(0));
+        assertTrue(method.isPresent());
+
+        List<String> tokens = methodNormalizer.normalize(method.get());
+        int expectedTokensSize = 66;
+
+        assertEquals(expectedTokensSize, tokens.size());
+
+//      FOR, LPAREN, INT, IDENTIFIER, ASSIGN, LITERAL, SEMI,
```

Comment:
Why are these comments in the tests, could you delete it or is it necessary?

Reply:
The test is a method normalizer using tokens to reduce noise during duplicate code analysis, I put those comments to show the tokens that are being generated based on a statement